streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -78,14 +78,24 @@
 
     private final Serde<K> keySerde;
     private final Serde<V> valSerde;
+    private final String storeName;
 
     private boolean sendOldValues = false;
 
+
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
                       Set<String> sourceNodes) {
-        this(topology, name, processorSupplier, sourceNodes, null, null);
+        this(topology, name, processorSupplier, sourceNodes, null, null, null);
+    }
+
+    public KTableImpl(KStreamBuilder topology,
```

Comment:
Thanks @guozhangwang. I just addressed 1 and 4 with the latest PR. Wanted to double check:
- For 3 (aggregation): did you find something missing? I think the above already passes the store name.
- Similarly for 2 (materialization): I believe I pass the store name. Please let me know what you think. Thanks.